### DISCUSSION TITLE
Gracefully Shutdown ASP.NET Core Web Host

### DISCUSSION THREAD

**Discussion by TonyValenti** [dotnet/runtime#68670] [Q&A]
Is there a way to gracefully shutdown an ASP.NET core Web Host so that active requests are serviced and the proper content is returned to the client?

I am doing some tests and it seems like calling ```IHostApplicationLifetime.StopApplication()``` results in some clients getting ```The response ended prematurely```.  What I'd like to make sure of is that any HTTP connections are fully serviced and receive proper responses (not even 500 errors).

Any thoughts?

**Comment by huoyaoyuan** [comment]
You could figure out whether ASP.NET or Kestrel provided a state that no longer accept new HTTP requests.

If such technique is deployed in production environment, I can see DoS attack by sending an incomplete request or not reading response.
